### PR TITLE
Bugfix/pep448 in op unpack fix

### LINKED ISSUE #3938: [BUG] Multiple starred expressions in a list
Hi !
I think Cython is not able to compile when I unpack multiple expressions separated by commas, in one list.

**Describe the bug**
When I want to compile my python file containing a condition like 
```python
if arr['name'] not in [*foo.bar.keys(), *foo.bar2.keys()]:
```

I get two error messages
```
Error compiling Cython file:
------------------------------------------------------------

if arr['name'] not in [*foo.bar.keys(), *foo.bar2.keys()]:
                            ^
------------------------------------------------------------

src/package/foo/bar.py:964:83: starred expression is not allowed here

Error compiling Cython file:
------------------------------------------------------------

if arr['name'] not in [*foo.bar.keys(), *foo.bar2.keys()]:
                                             ^
------------------------------------------------------------

src/package/foo/bar.py:964:83: starred expression is not allowed here
````

I know it is a very particular method and I could do 
```python
if arr['name'] not in [*foo.bar.keys()] + [*foo.bar2.keys()]:
```
But maybe it's possible to add this feature to Cython ?

I found this [issue](https://github.com/cython/cython/issues/2939) which looks like my problem, but setting different language_level doesn't change anything.

**To Reproduce**
To reproduce behaviour, try to compile 
```python
if arr['name'] not in [*foo.bar.keys(), *foo.bar2.keys()]:
```

**Expected behavior**
Errors should appear

**Environment (please complete the following information):**
 - OS: Linux & Windows
 - Python version 3.7.9
 - All Cython versions tested, even master branch

**Additional context**

### PR DESCRIPTION
**Pull request by Arvin6**
Ignore optimization when the unpacking operators are present  [[PEP - 448] ](https://www.python.org/dev/peps/pep-0448/)

Eg:
```
>>> a = [1,2,3] 
>>> b = [4,5,6]
>>> [*a]
[1,2,3]
>>> 6 in [*a, *b]
True
```
Thanks @scoder @da-woods for helping out with this contribution.

Fixes #3938

### CODE REVIEW

File: tests/run/pep448_extended_unpacking.pyx [hunk 1]
```diff
@@ -272,6 +272,24 @@ def unpack_list_keep_originals(a, b, c):
     return [*a, *b, 2, *c]
 
 
+def t3938(x, l, m):
```
Comment (scoder) [inline tests/run/pep448_extended_unpacking.pyx:275]: I'd rather have an explanatory name here. Having to understand that this refers to a GH ticket, and then looking that up, is too much when I just want to know the purpose of this test.

File: Cython/Compiler/Optimize.py [hunk 2]
```diff
@@ -1440,6 +1440,11 @@ def visit_PrimaryCmpNode(self, node):
             # note: lhs may have side effects
             return node
 
+        if any([arg.is_starred for arg in args]):
+            # note: Ignore optimisations if there is at least 
+            # one starred expression in the args
```
Comment (scoder) [inline Cython/Compiler/Optimize.py:1445]: This is a redundant comment. The code does exactly what the comment says. Let's make it a bit more interesting by giving an actual reason.
```suggestion
            # Starred arguments do not directly translate to comparisons or "in" tests.
```

**Review by scoder — COMMENTED** [review]
Thanks, that looks reasonable.

File: tests/run/pep448_extended_unpacking.pyx (hunk 1 above)
Comment (Arvin6) [reply tests/run/pep448_extended_unpacking.pyx:275]: Sounds very reasonable, on it.

Comment (Arvin6) [reply tests/run/pep448_extended_unpacking.pyx]: Done

**Comment by scoder** [comment]
Picked over to 0.29.x here:
https://github.com/cython/cython/commit/1faf51f0811ea49b25280957fb84ac731ac16320

**Comment by scoder** [comment]
Thanks

**Pull request merged**